File: quick sort/quickSort.cs
using System;

public class Program {
    public static void Main(string[] args) {
        void quickSort(int[] arr, int low, int high) {
            if (low < high) {
                int p = partition(arr, low, high);
                quickSort(arr, low, p - 1);
                quickSort(arr, p + 1, high);
            }
        }

        int partition(int[] arr, int low, int high) {
            int pivot = arr[high];
            int i = low - 1;
            for (int j = low; j < high; j++) {
                if (arr[j] <= pivot) {
                    i++;
                    swap(arr, i, j);
                }
            }
            swap(arr, i + 1, high);
            return i + 1;
        }
        
        void swap(int[] arr, int i, int j) {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }

        int[] arr = { 5, 2, 4, 6, 1, 3 };
        quickSort(arr, 0, arr.Length - 1);
        foreach (int i in arr) {
            Console.Write(i + " ");
        }
    }
}

File: quick sort/quickSort.js
function quickSort(arr, low, high) {
    if (low < high) {
        let pivot = partition(arr, low, high);
        quickSort(arr, low, pivot - 1);
        quickSort(arr, pivot + 1, high);
    }
}

function partition(arr, low, high) {
    let pivot = arr[high];
    let i = low -1;
    for (let j = low; j < high; j++) {
        if (arr[j] <= pivot) {
            i++;
            swap(arr, i, j);
        }
    }
    swap(arr, i + 1, high);
    return i + 1;
}

function swap(arr, i, j) {
    let temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

let list = [5, 2, 1, 8, 4];
quickSort(list, 0, list.length - 1);
console.log(list);